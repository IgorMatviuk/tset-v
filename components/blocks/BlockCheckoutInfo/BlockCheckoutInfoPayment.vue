<template>
  <div class="checkout-info">
    <div class="checkout-info__title">
      {{ $t('checkoutPage.paymentMethod') }}
    </div>
    <BFormGroup class="checkout-info__group">
      <BFormRadioGroup
        v-model="value.paymentInfo.paymentMethod.$model"
        class="checkout-info__radio-group"
        name="payment-method"
      >
        <BFormRadio
          v-if="isAbroadDelivery"
          :class="{ error: value.paymentInfo.paymentMethod.$error }"
          class="checkout-info__radio checkout-info__radio--payment"
          value="cod"
        >
          {{ $t('checkoutPage.paymentMethods.cod') }}
        </BFormRadio>
        <BFormRadio
          v-if="isAbroadDelivery"
          :class="{ error: value.paymentInfo.paymentMethod.$error }"
          class="checkout-info__radio checkout-info__radio--payment"
          value="privat24"
        >
          {{ $t('checkoutPage.paymentMethods.privat') }}
        </BFormRadio>
        <BFormRadio
          :class="{ error: value.paymentInfo.paymentMethod.$error }"
          class="checkout-info__radio checkout-info__radio--payment"
          value="liqPay"
        >
          Credit Card / Debit Card (LiqPay)
        </BFormRadio>
      </BFormRadioGroup>
    </BFormGroup>
  </div>
</template>

<script>
export default {
  props: {
    value: {
      type: Object,
      required: true
    },
    isAbroadDelivery: {
      type: Boolean,
      required: true
    }
  }
}
</script>

<style lang="scss">
.checkout-info {
  &__radio {
    &--payment {
      width: 100%;

      &:not(:last-of-type) {
        margin-right: 0;
        margin-bottom: 12px;
      }
    }
  }

  @include media-breakpoint-down(sm) {
    &__radio {
      &--payment {
        &:not(:last-of-type) {
          margin-bottom: 8px;
        }
      }
    }
  }
}
</style>
