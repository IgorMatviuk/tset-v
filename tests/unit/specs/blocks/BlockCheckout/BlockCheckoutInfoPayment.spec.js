import { mount, createLocalVue } from '@vue/test-utils'
import { BFormGroup, BFormRadioGroup, BFormRadio } from 'bootstrap-vue'

import BlockCheckoutInfoPayment from '@/components/blocks/BlockCheckoutInfo/BlockCheckoutInfoPayment'

const localVue = createLocalVue()
localVue.component('BFormGroup', BFormGroup)
localVue.component('BFormRadioGroup', BFormRadioGroup)
localVue.component('BFormRadio', BFormRadio)

describe('BlockCheckoutInfoPayment', () => {
  function createCheckoutInfoPayment(isAbroadDelivery) {
    const wrapper = mount(BlockCheckoutInfoPayment, {
      localVue,
      propsData: {
        value: {
          paymentInfo: {
            paymentMethod: {
              $model: ''
            }
          }
        },
        isAbroadDelivery
      },
      mocks: {
        $t: (str) => str
      }
    })

    return wrapper
  }

  test('using "abroad" delivery method', () => {
    const wrapper = createCheckoutInfoPayment(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('using non "abroad" delivery method', () => {
    const wrapper = createCheckoutInfoPayment(false)
    expect(wrapper.element).toMatchSnapshot()
  })
})
