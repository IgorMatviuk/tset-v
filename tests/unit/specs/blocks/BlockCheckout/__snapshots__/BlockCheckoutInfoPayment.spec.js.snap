// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockCheckoutInfoPayment using "abroad" delivery method 1`] = `
<div
  class="checkout-info"
>
  <div
    class="checkout-info__title"
  >
    
    checkoutPage.paymentMethod
  
  </div>
   
  <fieldset
    class="form-group checkout-info__group"
  >
    <!---->
    <div
      class="bv-no-focus-ring"
      role="group"
      tabindex="-1"
    >
      <div
        class="checkout-info__radio-group bv-no-focus-ring"
        role="radiogroup"
        tabindex="-1"
      >
        <div
          class="checkout-info__radio checkout-info__radio--payment custom-control custom-control-inline custom-radio"
        >
          <input
            autocomplete="off"
            class="custom-control-input"
            name="payment-method"
            type="radio"
            value="cod"
          />
          <label
            class="custom-control-label"
          >
            
        checkoutPage.paymentMethods.cod
      
          </label>
        </div>
         
        <div
          class="checkout-info__radio checkout-info__radio--payment custom-control custom-control-inline custom-radio"
        >
          <input
            autocomplete="off"
            class="custom-control-input"
            name="payment-method"
            type="radio"
            value="privat24"
          />
          <label
            class="custom-control-label"
          >
            
        checkoutPage.paymentMethods.privat
      
          </label>
        </div>
         
        <div
          class="checkout-info__radio checkout-info__radio--payment custom-control custom-control-inline custom-radio"
        >
          <input
            autocomplete="off"
            class="custom-control-input"
            name="payment-method"
            type="radio"
            value="liqPay"
          />
          <label
            class="custom-control-label"
          >
            
        Credit Card / Debit Card (LiqPay)
      
          </label>
        </div>
      </div>
      <!---->
      <!---->
      <!---->
    </div>
  </fieldset>
</div>
`;

exports[`BlockCheckoutInfoPayment using non "abroad" delivery method 1`] = `
<div
  class="checkout-info"
>
  <div
    class="checkout-info__title"
  >
    
    checkoutPage.paymentMethod
  
  </div>
   
  <fieldset
    class="form-group checkout-info__group"
  >
    <!---->
    <div
      class="bv-no-focus-ring"
      role="group"
      tabindex="-1"
    >
      <div
        class="checkout-info__radio-group bv-no-focus-ring"
        role="radiogroup"
        tabindex="-1"
      >
        <!---->
         
        <!---->
         
        <div
          class="checkout-info__radio checkout-info__radio--payment custom-control custom-control-inline custom-radio"
        >
          <input
            autocomplete="off"
            class="custom-control-input"
            name="payment-method"
            type="radio"
            value="liqPay"
          />
          <label
            class="custom-control-label"
          >
            
        Credit Card / Debit Card (LiqPay)
      
          </label>
        </div>
      </div>
      <!---->
      <!---->
      <!---->
    </div>
  </fieldset>
</div>
`;
